internal/conditions/conditions.go

```diff
@@ -4,8 +4,25 @@ import (
 	"fmt"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+func GetStatusConditionForOperationResult(result controllerutil.OperationResult, objectName string, err error) metav1.Condition {
```

Comment:
nice

Reply:
All credit to @mitchnielsen, I just moved it here to avoid an extra file